tasks/main.yml

```diff
@@ -39,14 +39,44 @@
     state: started
     enabled: true
 
-- name: Get the active Tuned profile
+- name: Get the active Tuned profiles
   command: tuned-adm active
   changed_when: false
-  register: __mssql_tuned_active_profile
+  register: __mssql_tuned_active_profiles
 
-- name: Set the mssql Tuned profile
-  command: tuned-adm profile mssql
-  when: '"mssql" not in __mssql_tuned_active_profile.stdout'
+# adding the mssql profile to the end of the list ensures
+# that it overrides conflicting settings in other profiles
+- name: Add mssql to the list of Tuned profiles
+  when: '"mssql" not in __mssql_tuned_active_profiles.stdout'
+  block:
+    - name: Add mssql to the list of Tuned profiles
+      command: >-
+        tuned-adm profile {{ __mssql_tuned_active_profiles.stdout |
+        regex_replace( '^Current active profile: ', '' ) }} mssql
+      register: __mssql_tuned_adm_profile
+
+  # Rescue is needed because there is a bug in tuned that causes issues when
+  # adding multiple profiles with common ancestors. Fail happens for example,
+  # when running `tuned-adm profile virtual-guest mssql` because both profiles
+  # include `throughput-performance`
+  # https://bugzilla.redhat.com/show_bug.cgi?id=1825882
+  rescue:
+    - name: Remove troublemaking include from the mssql profile
+      lineinfile:
+        path: /usr/lib/tuned/mssql/tuned.conf
+        regexp: include=throughput-performance
+        state: absent
+      when: >-
+        "Cannot find profile 'throughput-performance'" in
+        __mssql_tuned_adm_profile.stderr
```

Comment:
can you use `ansible_failed_result.results.0.msg` here instead of `register: __mssql_tuned_adm_profile`?